common/datavalues/src/vec.rs

```diff
@@ -11,31 +11,32 @@ use common_arrow::arrow::alloc;
 use common_arrow::arrow::array::ArrayData;
 use common_arrow::arrow::array::PrimitiveArray;
 use common_arrow::arrow::buffer::Buffer;
+use common_arrow::arrow::buffer::MutableBuffer;
 use common_arrow::arrow::datatypes::*;
 
 /// A `Vec` wrapper with a memory alignment equal to Arrow's primitive arrays.
 /// Can be useful in creating a new DataArray or Arrow Primitive array without copying.
 #[derive(Debug)]
-pub struct AlignedVec<T> {
+pub struct AlignedVec<T: ArrowNativeType> {
     pub inner: Vec<T>,
     // if into_inner is called, this will be true and we can use the default Vec's destructor
     taken: bool,
 }
 
-impl<T> Drop for AlignedVec<T> {
+impl<T: ArrowNativeType> Drop for AlignedVec<T> {
     fn drop(&mut self) {
         if !self.taken {
             let inner = mem::take(&mut self.inner);
             let mut me = mem::ManuallyDrop::new(inner);
             let ptr: *mut T = me.as_mut_ptr();
             let ptr = ptr as *mut u8;
             let ptr = std::ptr::NonNull::new(ptr).unwrap();
-            unsafe { alloc::free_aligned::<u8>(ptr, self.capacity()) }
+            unsafe { alloc::free_aligned::<u8>(ptr, me.capacity() * mem::size_of::<T>()) }
```

Comment:
So the previous implementation has bug? Because `self.capacity()` is not the size.

Reply:
Is there any tool like github actions help us to check the  unsafe?

Reply:
> So the previous implementation has bug? Because `self.capacity()` is not the size.

Known UB.

> Is there any tool like github actions help us to check the unsafe?

cargo miri, but some preparation is needed, we can open an issue tracking.